src/game_inventory.cpp

```diff
@@ -534,22 +534,36 @@ class comestible_inventory_preset : public inventory_selector_preset
+            const std::string nourishment_title = _( "NOURISHMENT" );
+            append_cell( [&p, &nourishment_title]( const item_location & loc ) {
+                /* Understanding how Calories Per Effective Volume are calculated requires a dive into the
+                stomach fullness source code. Look at issue #44365*/
                 const item &it = *loc;
-                const int charges = std::max( it.charges, 1 );
-                const double converted_weight = convert_weight( it.weight() / charges );
-                if( converted_weight == 0 ) {
-                    return std::string( "---" );
-                }
-                const nutrients nutr = p.compute_effective_nutrients( *loc );
+                const nutrients nutr = p.compute_effective_nutrients( it );
                 const int kcalories = nutr.kcal;
-                // Experimental: if calories are 0 (medicine, batteries etc), don't display anything.
-                if( kcalories == 0 ) {
+                // Quit prematurely if there is no caloric content/the item is not food.
+                if( kcalories == 0 || !it.type->comestible ) {
                     return std::string();
                 }
-                const int calpergr = int( std::round( kcalories / converted_weight ) );
-                return string_format( _( "%d" ), calpergr );
-            }, _( "CAL/kg" ) );
+                units::volume water_vol = ( it.type->comestible->quench > 0 ) ? it.type->comestible->quench *
+                                          5_ml : 0_ml;
+                // Water volume is ignored.
+                units::volume food_vol = it.volume() - water_vol;
+                const double converted_volume = round_up( convert_volume( food_vol.value() / it.count() ), 2 );
+                const double energy_density_ratio = p.compute_effective_food_volume_ratio( it );
+                const double effective_volume = converted_volume * energy_density_ratio;
+                const int calories_per_effective_volume = std::round( kcalories / effective_volume );
+                // Arbitrary max value we will cap our vague display to. Will be lower than the actual max value, but scaling fixes that.
+                constexpr int max_cal_per_effective_vol = 1500;
+                const int scaled_max = std::sqrt( max_cal_per_effective_vol ) / 4;
+                const int scaled_cal = std::sqrt( calories_per_effective_volume ) / 4;
+                const std::pair<std::string, nc_color> nourishment_bar = get_bar(
+                            scaled_cal, scaled_max, 5, true );
+                // This colorizes the bar, after padding it to 5 spaces using dots and up to nourishment_title.length() using spaces, to preserve rtl alignment.
+                return colorize( nourishment_bar.first + std::string( 5 - nourishment_bar.first.length(),
+                                 '.' ) + std::string( nourishment_title.length() - 5, ' ' ), nourishment_bar.second );
```

Comment:
`nourishment_title.length()` isn't the right thing to be using here.  You need to use `utf8_width( nourishment_title )`.

Also, maybe split this up into multiple statements rather than one long, messy expression.
e.g.
```c++
std::string result = nourishment_bar.first;
result += std::string( 5 - nourishment_bar.first.length(), '.' );
result += std::string( utf8_width( nourishment_title ) - 5, ' ' );
return colorize( result, nourishment_bar.second );
```

Reply:
Thanks for the feedback. I have never touched localization stuff before. I will clean up the code and add a case where if the utf8 width of the localized nourishment bar is less than 5, no padding is done. The way it is right now, it will probably crash instead.

Reply:
Hm. If I pass nourishment_title by reference, it's utf8_width is always 0. I have to pass it by copy in order to get the proper length.